pkg/api/upgrade.go

```diff
@@ -1,129 +1,51 @@
 package api
 
 import (
-	"context"
 	"encoding/json"
-	"fmt"
+	"io"
 	"net/http"
 
-	"helm.sh/helm/v3/pkg/action"
-	"helm.sh/helm/v3/pkg/chart/loader"
-	"helm.sh/helm/v3/pkg/cli/values"
-	"helm.sh/helm/v3/pkg/getter"
-	"helm.sh/helm/v3/pkg/servercontext"
-	"helm.sh/helm/v3/pkg/storage/driver"
+	"helm.sh/helm/v3/pkg/api/logger"
 )
 
 type UpgradeRequest struct {
-	RequestID        string
-	ReleaseName      string
-	ReleaseNamespace string
-	ChartPath        string
+	Name      string                 `json:"name"`
+	Namespace string                 `json:"namespace"`
+	Chart     string                 `json:"chart"`
+	Values    map[string]interface{} `json:"values"`
 }
 
 type UpgradeResponse struct {
-	Status        bool
-	ReleaseStatus string
+	Error  string `json:"error,omitempty"`
+	Status string `json:"status,omitempty"`
 }
 
-type UpgradeHandler struct {
-	service Service
-}
-
-func Upgrade(service Service) http.Handler {
-	return UpgradeHandler{service: service}
-}
-
-func (h UpgradeHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
-	res.Header().Set("Content-Type", "application/json")
-	defer req.Body.Close()
-
-	var request UpgradeRequest
-	decoder := json.NewDecoder(req.Body)
-	decoder.UseNumber()
+func Upgrade(svc Service) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
```

Comment:
move this as mux middleware, as it is common for all handlers

Reply:
Done